src/main/python/tests/examples/tutorials/test_adult.py

```diff
@@ -0,0 +1,324 @@
+
+        X1 = F1.rbind(F2)
+        X1, M1 = X1.transform_encode(spec=jspec)
+
+        X = PREPROCESS_package.get_X(X1, 1, train_count)
+        Y = PREPROCESS_package.get_Y(X1, 1, train_count)
+
+        Xt = PREPROCESS_package.get_X(X1, train_count, train_count + test_count)
+        Yt = PREPROCESS_package.get_Y(X1, train_count, train_count + test_count)
+
+        Yt = PREPROCESS_package.replace_value(Yt, 3.0, 1.0)
+        Yt = PREPROCESS_package.replace_value(Yt, 4.0, 2.0)
+
+        # better alternative for encoding
+        # F2 = F2.replace("<=50K.", "<=50K")
+        # F2 = F2.replace(">50K.", ">50K")
+        # X1, M = F1.transform_encode(spec=jspec)
+        # X2 = F2.transform_apply(spec=jspec, meta=M)
+
+        # X = PREPROCESS_package.get_X(X1, 1, train_count)
+        # Y = PREPROCESS_package.get_Y(X1, 1, train_count)
+        # Xt = PREPROCESS_package.get_X(X2, 1, test_count)
+        # Yt = PREPROCESS_package.get_Y(X2, 1, test_count)
+
+        # TODO somehow throws error at predict with this included
+        # X, mean, sigma = scale(X, True, True)
+        # Xt = scaleApply(Xt, mean, sigma)
+
+        FFN_package = self.sds.source(self.neural_net_src_path, "fnn", print_imported_methods=True)
+
+        epochs = 1
+        batch_size = 16
+        learning_rate = 0.01
+        seed = 42
+
+        network = FFN_package.train(X, Y, epochs, batch_size, learning_rate, seed)
+
+        """
+        If more ressources are available, one can also choose to train the model using a parameter server.
+        Here we use the same parameters as before, however we need to specifiy a few more.
+        """
+        ################################################################################################################
+        # workers = 1
+        # utype = '"BSP"'
+        # freq = '"EPOCH"'
+        # mode = '"LOCAL"'
+        # network = FFN_package.train_paramserv(X, Y, epochs,
+        #                                       batch_size, learning_rate, workers, utype, freq, mode,
+        #                                       seed)
+        ################################################################################################################
+
+        FFN_package.save_model(network, '"model/python_FFN/"').compute(verbose=True)
+
+        """
+        Next we evaluate our network on the test set which was not used for training.
+        The predict function with the test features and our trained network returns a matrix of class probabilities.
+        This matrix contains for each test sample the probabilities for each class.
+        For predicting the most likely class of a sample, we choose the class with the highest probability.
+        """
+        ################################################################################################################
+        #probs = FFN_package.predict(Xt, network)
```

Comment:
this was not working, correct?

Reply:
Yes, correct (predict seems to be the problem).